cylc/uiserver/schema.py

```diff
@@ -254,6 +259,227 @@ class Arguments:
     result = GenericScalar()
 
 
+async def get_jobs(root, info, **args):
+    if args['live']:
+        return await get_nodes_all(root, info, **args)
+
+    _, field_ids = process_resolver_info(root, info, args)
+
+    if hasattr(args, 'id'):
+        args['ids'] = [args.get('id')]
+    if field_ids:
+        if isinstance(field_ids, str):
+            field_ids = [field_ids]
+        elif isinstance(field_ids, dict):
+            field_ids = list(field_ids)
+        args['ids'] = field_ids
+    elif field_ids == []:
+        return []
+
+    for arg in ('ids', 'exids'):
+        # live objects can be represented by a universal ID
+        args[arg] = [Tokens(n_id, relative=True) for n_id in args[arg]]
+    args['workflows'] = [
+        Tokens(w_id) for w_id in args['workflows']]
+    args['exworkflows'] = [
+        Tokens(w_id) for w_id in args['exworkflows']]
+
+    return await list_jobs(args)
+
+
+async def list_jobs(args):
+    if not args['workflows']:
+        raise Exception('At least one workflow must be provided.')
+    from cylc.flow.workflow_db_mgr import WorkflowDatabaseManager
+    # from cylc.flow.rundb import CylcWorkflowDAO
+    from pathlib import Path
+    from cylc.flow.workflow_files import get_workflow_srv_dir
+
+    jobs = []
+    for workflow in args['workflows']:
+        db_file = Path(get_workflow_srv_dir(workflow['workflow']), 'db')
+        wdbm = WorkflowDatabaseManager(db_file.parent)
+        with wdbm.get_pri_dao() as pri_dao:
+            conn = pri_dao.connect()
+            jobs = make_query(conn, workflow)
```

Comment:
> I'm not familiar with the inner workings of cylc. But your suggestion doesn't work for me.
You sure this is correct?

Oh, one thing was missing was specifying the public DB path instead of the private path. (The private path is only meant to be used by the scheduler; the public path is for 3rd(ish) party stuff like this)

This works for me

```suggestion
    from cylc.flow.rundb import CylcWorkflowDAO
    from cylc.flow.pathutil import get_workflow_run_dir
    from cylc.flow.workflow_files import WorkflowFiles

    jobs = []
    for workflow in args['workflows']:
        db_file = get_workflow_run_dir(
            workflow['workflow'],
            WorkflowFiles.LogDir.DIRNAME,
            WorkflowFiles.LogDir.DB
        )
        with CylcWorkflowDAO(db_file, is_public=True) as dao:
            conn = dao.connect()
            jobs.extend(make_query(conn, workflow))
```

Reply:
I've had a look at git blame and it was added in cylc-flow 8.1.1. May be best to replace it with `'db'` for backwards compatibility